pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -290,7 +292,18 @@ def _do_catch(self, pokemon, encounter_id, catch_rate_by_ball, is_vip=False):
                 current_ball += 1
             if items_stock[current_ball] == 0:
                 self.emit_event('no_pokeballs', formatted='No usable pokeballs found!')
-                break
+                if maximum_ball != ITEM_ULTRABALL and items_stock[ITEM_ULTRABALL] > min_ultraball_to_keep:
+                    maximum_ball = ITEM_ULTRABALL
+                    self.emit_event(
+                        'use_ultraball',
+                        formatted='Use ultraball instead (ultraball count > {ultraball_count})',
+                        data={
+                            'ultraball_count': min_ultraball_to_keep
+                        }
+                    )
+                    continue
```

Comment:
Exactly, `continue` will go back to the `while True`, but the different is `maximum_ball` is set to `ITEM_ULTRABALL` while it was `ITEM_GREATBALL` before the `while True`. That's the point.